libs/pages/database/src/lib/feature/page-settings-general-feature/page-settings-general-feature.tsx

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from 'react'
+import { FieldValues, FormProvider, useForm } from 'react-hook-form'
+import { useDispatch, useSelector } from 'react-redux'
+import { useParams } from 'react-router-dom'
+import { editDatabase, selectDatabaseById } from '@console/domains/database'
+import { DatabaseEntity } from '@console/shared/interfaces'
+import { AppDispatch, RootState } from '@console/store/data'
+import PageSettingsGeneral from '../../ui/page-settings-general/page-settings-general'
+
+export const handleSubmit = (data: FieldValues, database: DatabaseEntity) => {
+  const cloneDatabase = Object.assign({}, database as DatabaseEntity)
+  cloneDatabase.name = data['name']
+  cloneDatabase.accessibility = data['accessibility']
+
+  return cloneDatabase
+}
+
+export function PageSettingsGeneralFeature() {
+  const { databaseId = '' } = useParams()
+  const dispatch = useDispatch<AppDispatch>()
+  const database = useSelector<RootState, DatabaseEntity | undefined>((state) => selectDatabaseById(state, databaseId))
+
+  const [loading, setLoading] = useState(false)
+
+  const methods = useForm({
+    mode: 'onChange',
+    reValidateMode: 'onChange',
+  })
+
+  const onSubmit = methods.handleSubmit((data) => {
+    if (data && database) {
+      setLoading(true)
+      const cloneDatabase = handleSubmit(data, database)
+
+      dispatch(
+        editDatabase({
+          databaseId: databaseId,
+          data: cloneDatabase,
+        })
+      )
+        .unwrap()
+        .then(() => setLoading(false))
+        .catch((e) => {
+          setLoading(false)
+          console.error(e)
+        })
+    }
+  })
+
+  useEffect(() => {
+    methods.setValue('name', database?.name)
```

Comment:
Using a `methods.reset` is often better than using multiple setValue for many reasons. setValue is working here so you can let like but just to let you know, you can quickly read this answer:
https://github.com/react-hook-form/react-hook-form/discussions/7671#discussioncomment-2052400

The part about resetting the defaultValues is something I needed on another of my component last time. I don't quite remember why though.

Reply:
I think it was especially important in the case we want to use `value` instead of `defaultValue`